os/drivers/pm/pm.c

```diff
@@ -92,15 +97,31 @@ static int pm_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	/* Handle built-in ioctl commands */
 	switch (cmd) {
     	case PMIOC_SUSPEND:
-	        pm_stay(((pm_arg_t *)arg)->domain, ((pm_arg_t *)arg)->state);
+	        pm_stay(PM_IDLE_DOMAIN, (enum pm_state_e)arg);
 		pmvdbg("State locked!\n");
 		ret = OK;
         	break;
         case PMIOC_RESUME:
-		pm_relax(((pm_arg_t *)arg)->domain, ((pm_arg_t *)arg)->state);
+		pm_relax(PM_IDLE_DOMAIN, (enum pm_state_e)arg);
 		pmvdbg("State unlocked!\n");
 		ret = OK;
 		break;
+	case PMIOC_SLEEP:
+		if (arg > 0) {
+			/* Converting from milliseconds to ticks */
+			arg = ((arg * CLOCKS_PER_SEC) /  1000);
```

Comment:
Do we not need to provide an API in milliseconds for kernel functions that use pm_sleep rather than driver?

I think it is better to do this in the pm_sleep() function. And if necessary, please change the name to pm_msleep().